Hisat-TLA is under development!
============

What is HISAT-TLA?
-----------------

HISAT-TLA is a three letter aligner(TLA) integrated in HISAT2. It could be used for any base-converted sequencing reads like
slam-seq, bisulfite-seq.



Build index for HISAT-TLA
============

mkRef
-------------
HISAT-TLA need two HISAT2 index as input. Before we build HISAT2 index, we need to use 'mkRef' to make converted reference.

mkRef usage:
* `--reference`  
    The reference input file (FASTA file). 
    
* `--base-change`  
    Provide which base is converted in sequencing process to other base. Please enter
    2 letter for this argument. First letter should be the converted from base, second letter should be
    the converted to base. For example, during slam-seq, some 'T' is converted to 'C',
    please enter `--base-change TC`. During bisulfite-seq, some 'C' is converted to 'T',please enter `--base-change CT`.
  
* `--o1`  
    The output filename for 1st converted reference generated by mkRef (FASTA file).
     
* `--o2`  
    The output filename for 2nd converted reference generated by mkRef (FASTA file).

Sample argument for slam-seq reference preparation:  
`mkRef -reference genome.fa --base-change TC --o1 genome_TC.fa --o2 genome_AG.fa`

hisat2-build
-----------
After you generated 2 converted reference file by mkRef, you need to run hisat2-build 2 times to build HISAT-TLA index.
Please make sure you know which index is built from 1st converted reference.

Sample argument for slam-seq index building:  
`hisat2-build genome_TC.fa genome_TC -p 10`   
`hisat2-build genome_AG.fa genome_AG -p 10` 

HISAT-TLA index with splicing site information is under development!  
PLEASE DO NOT USE THIS!

    HISAT2_SS_SCRIPT=./hisat2_extract_splicesites.py
    HISAT2_EXON_SCRIPT=./hisat2_extract_exons.py
    download ftp://ftp.ensembl.org/pub/release-84/gtf/homo_sapiens/Homo_sapiens.GRCh38.84.gtf.gz
    
    tar zxvf Homo_sapiens.GRCh38.84.gtf.gz
    mv  Homo_sapiens.GRCh38.84.gtf genome.gtf
    GTF_FILE=genome.gtf
    ${HISAT2_SS_SCRIPT} ${GTF_FILE} > genome.ss
    ${HISAT2_EXON_SCRIPT} ${GTF_FILE} > genome.exon

Sample argument for slam-seq index building with splicing site information:  
`hisat2-build genome_TC.fa genome_TC -p 10 --ss genome.ss --exons genome.exon`   
`hisat2-build genome_AG.fa genome_AG -p 10 --ss genome.ss --exons genome.exon` 


Use HISAT-TLA for alignment
============
After we build 2 HISAT2 index, you are ready to use HISAT-TLA for alignemnt.
HISAT-TLA use HISAT2 argument but has some extra argument.

For human genome reference, it took about 15GB for HISAT-TLA alignment.

* `--TLA`  
    Use TLA mode. If you do not enter this argument, HISAT-TLA will run like HISAT2.

* `--base-change`  
    Provide which base is converted in sequencing process to other base. Please enter
    2 letter for this argument. First letter should be the converted base, second letter should be
    the converted to base. For example, during slam-seq, some 'T' is converted to 'C',
    please enter `--base-change TC`. During bisulfite-seq, some 'C' is converted to 'T',please enter `--base-change CT`.
    Please make sure you enter extact same --base-change argument as mkRef.
    
* `--reference`
      The original genome reference file(fasta).
      Please make sure you enter extact same --reference argument as mkRef.
      
* `--index1`  
    The basename of the 1st converted index for the reference genome.  The basename is the name of
      any of the index files up to but not including the final `.1.ht2` / etc. Please make sure this index is built with
      --o1 output from mkRef.
      
* `--index2`  
    The basename of the 2nd converted index for the reference genome.  The basename is the name of
    any of the index files up to but not including the final `.1.ht2` / etc. Please make sure this index is built with
    --o2 output from mkRef.
    
* `--expand-repeat` 
    If you used repeat index (`--repeat`), this argument will help you expand the alignment result if it aligned to repeat index.    
          
* `--repeat-limit` 
    If you used repeat index and expand repeat `--repeat --expand-repeat` mode, you could set up number of alignment will be check
    for each repeat alignment. default: 1000.

Sample argument:  
* If you have repeat index for slam-seq:  
`--hisat2  --index1 genome_TC  --index2 genome_AG --reference genome.fa -f -1 read_1.fa  -2 read_2.fa -S output.sam --TLA --base-change TC --repeat -p 10 --expand-repeat`

* If you have non-repeat index for slam-seq:  
`--hisat2  --index1 genome_TC  --index2 genome_AG --reference genome.fa -f -1 read_1.fa  -2 read_2.fa -S output.sam --TLA --base-change TC -p 10`

